lib/galaxy/webapps/galaxy/api/genomes.py

```diff
@@ -43,3 +43,39 @@ def show( self, trans, id, num=None, chrom=None, low=None, high=None, **kwd ):
         else:
             rval = self.app.genomes.chroms( trans, dbkey=id, num=num, chrom=chrom, low=low )
         return rval
+
+    @web.expose_api_raw_anonymous
+    def indexes(self, trans, id, **kwd):
+        """
+        GET /api/genomes/{id}/indexes?type={table name}
+
+        Returns all available indexes for a genome id for type={table name}
+        For instance, /api/genomes/hg19/indexes?type=fasta_indexes
+        """
+        index_extensions = {'fasta_indexes': '.fai'}
+        id = get_id( id, kwd.get( 'format', None ) )
+        index_type = kwd.get('type', None)
+
+        tbl_entries = self.app.tool_data_tables.data_tables[index_type].data
+        index_file_name = [x[-1] for x in tbl_entries if id in x].pop()
+
+        if_open = open(index_file_name + index_extensions[index_type], mode='r')
+        return if_open.read()
+
+    @web.expose_api_raw_anonymous
+    def sequences(self, trans, id, num=None, chrom=None, low=None, high=None, **kwd ):
+        """
+        GET /api/genomes/{id}/sequences
+
+        This is a wrapper for accepting sequence requests that
+        want a raw return, not json
+        """
+        # import pdb
+        # pdb.set_trace()
```

Comment:
Tom McGowan @tmcgowan added a IGV.js widget to the proteomics visualization application.  
The goal is to have the visualization be able to evaluate MassSpec identified novel peptide sequences in a genome context along with corresponding RNAseq data.   
Tom McGowan added the api calls to genome.py so that IGV.js could request genomic sequence as needed rather than downloading and caching the entire genome fasta file.
This requires requires retrieving the fasta index.   
 He also made a couple minor modifications to IGV.js to support retrieving the reference sequence from a galaxy server, which also work for rest.ensembl.org.  
We plan on submitting a pull request to IGV.js when we've had time to do a little more testing.